std/bigint.d

```diff
@@ -277,8 +277,8 @@ public:
 
     // BigInt op integer
     BigInt opBinary(string op, T)(T y) pure nothrow const
-        if ((op=="+" || op == "*" || op=="-" || op=="/" || op=="|" || op=="&" ||
-            op=="^"|| op==">>" || op=="<<" || op=="^^")
+        if ((op=="+" || op == "*" || op=="-" || op=="/" || op=="%" || op=="|" ||
+            op=="&" || op=="^" || op==">>" || op=="<<" || op=="^^")
```

Comment:
> The interesting thing about modulo is that a BigInt modulo a long/ulong can return a long/ulong (the result must fit by definition of modulo). Can you separate out the implementation to do this?

A negative BigInt modulo a positive ulong will be negative, and the value may go out of the range of long/ulong. For example:
```d
BigInt x = BigInt(long.min) - 1;  // BigInt("-9223372036854775809")
assert(x < long.min);
ulong y = ulong.max;
auto z = x % y;
assert(is(typeof(z) == BigInt));
assert(z == x);  // no overflow!
assert(z.toLong() == -9223372036854775807L);  // overflow!
```

Therefore, I think the return value should be a BigInt to avoid complex behavior.

> Won't this definition conflict with the existing definition on line 289?

No. `T : int` of the existing one should match types other than long/ulong.